clifford/_multivector.py

```diff
@@ -651,33 +652,45 @@ def isScalar(self) -> bool:
 
         return True
 
-    def isBlade(self) -> bool:
+    def isBlade(self, *, invertible=False) -> bool:
         """Returns true if multivector is a blade.
         """
         if len(self.grades()) != 1:
             return False
 
-        return self.isVersor()
+        return self.isVersor(invertible=invertible)
 
-    def isVersor(self) -> bool:
+    def isVersor(self, *, invertible=True) -> bool:
         """Returns true if multivector is a versor.
         From Leo Dorsts GA for computer science section 21.5, definition from 7.6.4
         """
+        if invertible:
+            grade_cmp = operator.eq
+        else:
+            grade_cmp = operator.le  # subset
+
         Vhat = self.gradeInvol()
         Vrev = ~self
-        Vinv = Vrev/(self*Vrev)[0]
+        mag = (self*Vrev)[()]
+        if mag != 0:
+            # not strictly necessary, just scales to make eps appropriate below
+            Vinv = Vrev / mag
+        elif invertible:
+            return False
+        else:
+            Vinv = Vrev
 
         # Test if the versor inverse (~V)/(V * ~V) is truly the inverse of the
         # multivector V
-        if grades_present(Vhat*Vinv, 0.000001) != {0}:
+        if not grade_cmp(grades_present(Vhat*Vinv, 0.000001), {0}):
             return False
         if not np.sum(np.abs((Vhat*Vinv).value - (Vinv*Vhat).value)) < 0.0001:
             return False
 
         # applying a versor (and hence an invertible blade) to a vector should
         # not change the grade
         if not all(
-            grades_present(Vhat*e*Vrev, 0.000001) == {1}
+            grade_cmp(grades_present(Vhat*e*Vrev, 0.000001), {1})
             for e in cf.basis_vectors(self.layout).values()
         ):
```

Comment:
This condition is relaxed to allow only the removal of grades, but not the addition of any new ones